test/e2e/privilegesmgmt/ssr.go

```diff
@@ -73,16 +73,30 @@ func SSRTest() {
 			fmt.Sprintf("Failed to create an ssr object in %s namespace", VeleroCfg.VeleroNamespace))
 
 		ssrListResp := new(v1.ServerStatusRequestList)
-
 		By(fmt.Sprintf("Check ssr object in %s namespace", VeleroCfg.VeleroNamespace))
-		Expect(client.Kubebuilder.List(ctx, ssrListResp, &kbclient.ListOptions{Namespace: VeleroCfg.VeleroNamespace})).To(Succeed(),
-			fmt.Sprintf("Failed to list ssr object in %s namespace", VeleroCfg.VeleroNamespace))
-		Expect(len(ssrListResp.Items)).To(BeNumerically("==", 1),
-			fmt.Sprintf("Count of ssr object in %s namespace is not 1", VeleroCfg.VeleroNamespace))
-		Expect(ssrListResp.Items[0].Status.ServerVersion).NotTo(BeEmpty(),
-			fmt.Sprintf("ServerVersion of ssr object in %s namespace should not empty", VeleroCfg.VeleroNamespace))
-		Expect(ssrListResp.Items[0].Status.Phase == "Processed").To(BeTrue(),
-			fmt.Sprintf("Phase of ssr object in %s namespace should be Processed", VeleroCfg.VeleroNamespace))
+		err = waitutil.PollImmediateInfinite(5*time.Second,
+			func() (bool, error) {
+				if err = client.Kubebuilder.List(ctx, ssrListResp, &kbclient.ListOptions{Namespace: VeleroCfg.VeleroNamespace}); err != nil {
+					if err == context.DeadlineExceeded {
+						return false, fmt.Errorf("execeed test case deadline and failed to check ssr object in %s namespace", VeleroCfg.VeleroNamespace)
+					}
+					return false, fmt.Errorf("failed to list ssr object in %s namespace with err %v", VeleroCfg.VeleroNamespace, err)
+				}
+				if len(ssrListResp.Items) != 1 {
+					return false, fmt.Errorf("count of ssr object in %s namespace is not 1", VeleroCfg.VeleroNamespace)
+				}
+
+				if ssrListResp.Items[0].Status.ServerVersion == "" {
+					fmt.Printf("ServerVersion of ssr object in %s namespace should not empty, current response result %v\n", VeleroCfg.VeleroNamespace, ssrListResp)
```

Comment:
It's updated to `waitutil.PollImmediateInfinite(`, if the ServerVersion remains "", this will be an infinite loop?

Reply:
But if the context is exceed the deadline, it will break the loop

Reply:
It's quite confusing, I suggest using `PollImmediateI` and pass the timeout as a parm of the `PollImmediate` func.

Reply:
ok, I've changed to PollImmediate func

Reply:
thx!